libafl_nyx/src/executor.rs

```diff
@@ -0,0 +1,100 @@
+use libafl::{
+    executors::{Executor, ExitKind, HasObservers},
+    inputs::{HasBytesVec, Input},
+    observers::ObserversTuple,
+};
+use libnyx::NyxReturnValue;
+use std::fmt::Debug;
+/// executor for nyx
+use std::marker::PhantomData;
+
+use crate::helper::{NyxHelper, NyxResult};
+
+/// executor for nyx standalone mode
+pub struct NyxStandaloneExecutor<'a, I, S, OT> {
+    /// implement nyx function
+    pub helper: &'a mut NyxHelper,
+    /// observers
+    observers: OT,
+    /// phantom data to keep generic type <I,S>
+    phantom: PhantomData<(I, S)>,
+}
+
+impl<'a, I, S, OT> Debug for NyxStandaloneExecutor<'a, I, S, OT> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("NyxInprocessExecutor")
+            .field("helper", &self.helper)
+            .finish()
+    }
+}
+
+impl<'a, EM, I, S, Z, OT> Executor<EM, I, S, Z> for NyxStandaloneExecutor<'a, I, S, OT>
+where
+    I: Input + HasBytesVec,
+{
+    fn run_target(
+        &mut self,
+        _fuzzer: &mut Z,
+        _state: &mut S,
+        _mgr: &mut EM,
+        input: &I,
+    ) -> Result<libafl::executors::ExitKind, libafl::Error> {
+        let input = input.bytes();
+        self.helper.nyx_process.set_input(input, input.len() as u32);
+
+        // exec will take care of trace_bits, so no need to reset
+        let ret_val = self.helper.nyx_process.exec();
+        match ret_val {
+            NyxReturnValue::Normal => Ok(ExitKind::Ok),
+            NyxReturnValue::Crash | NyxReturnValue::Asan => Ok(ExitKind::Crash),
+            NyxReturnValue::Timeout => Ok(ExitKind::Timeout),
+            NyxReturnValue::InvalidWriteToPayload => {
+                println!("FixMe: Nyx InvalidWriteToPayload handler is missing");
+                Err(libafl::Error::ShuttingDown)
```

Comment:
maybe change this error type to error::Illegalstate(format!("FixMe: Nyx InvalidWriteToPayload handler is missing")) here